site/src/components/Carousel.js

```diff
@@ -0,0 +1,165 @@
+    var slidePaddingVar = 25;
+    var navButtonSizeVar = 50;
+    const Svg = require('@site/static/img/amoeba.svg').default;
+    const [currentIndex, setCurrentIndex] = useState(itemList.length * 2);
+    const [hasTransitionClass, setHasTransitionClass] = useState(true);
+    const [navDisabled, setNavDisabled] = useState(false);
+    const [slidePadding, setSlidePadding] = useState(slidePaddingVar);
+    const [itemWidth, setItemWidth] = useState(itemWidthVar);
+    const [itemHeight, setItemHeight] = useState(itemWidth * (2 / 3));
+    const [carouselWidth, setCarouselWidth] = useState(70);
+    const [navButtonSize, setNavButtonSize] = useState(navButtonSizeVar);
+    
+    var newItemList = itemList.concat(itemList, itemList, itemList);
+
+    var carouselContentStyle = {
+        width: `${carouselWidth}vw`
+    }
+    var carouselSlidesStyle = {
+        marginLeft: `-${slidePadding}px`,
+        transform: `translateX(-${currentIndex * (itemWidth + (slidePadding * 2))}px)`
+    }
+    var navButtonsStyle = {
+        transform: `translateX(-${(100 - carouselWidth) / 2}vw)`,
+        top: `calc(50% - ${navButtonSize / 2}px)`
+    }
+    var buttonStyle = {
+        width: `${navButtonSize}px`,
+        height: `${navButtonSize}px`
+    }
+    
+    const scrollCarousel = (newIndex) => {  // allow nav buttons to scroll carousel
+        setCurrentIndex(newIndex);
+    };
+
+    useEffect(() => {   // carousel dynamic resizing based on screen width
+        const resizeCarousel = () => {
+            if (window.innerWidth > 996) {
+                itemWidthVar = 380;
+                slidePaddingVar = 25;
+                navButtonSizeVar = 50;
+            }
+            if (window.innerWidth <= 996 && window.innerWidth > 480) {
+                itemWidthVar = 300;
+                slidePaddingVar = 25;
+                navButtonSizeVar = 50;
+            } 
+            if (window.innerWidth <= 480) {
+                itemWidthVar = carouselWidth * window.innerWidth / 100;
+                slidePaddingVar = 15;
+                navButtonSizeVar = 40;
+            } 
+            setItemWidth(itemWidthVar);
+            setItemHeight(itemWidthVar * (2 / 3));
+            setSlidePadding(slidePaddingVar);
+            setNavButtonSize(navButtonSizeVar);
+        }
+
+        window.addEventListener('resize', resizeCarousel);
+
+        return () => window.removeEventListener('resize', resizeCarousel);
+    }, [])
+
+    useEffect(() => {   // make carousel infinite
+        if (currentIndex > itemList.length * 2) {   // keep index near the middle of the list when moving left
+            setNavDisabled(true);   // disable nav buttons while silently moving to different card
+            setTimeout(() => {  // disable transition animation after card slides, then snap to same card prior in the list
+                setHasTransitionClass(false);
+                setCurrentIndex(currentIndex - itemList.length);
+            }, 250)
```

Comment:
Where is this 250 coming from?  It seems to be repeated, could we extract this as a constant with more context as to what this value is doing?

Reply:
That's the timeout in ms that we wait for the carousel to wrap around. I can make that a variable that gets reused.